### PR TITLE
MIR-1260 Allow to generate svg image indicating file type

### PR DESCRIPTION
**Pull request by Possommi**
[Link to jira](https://mycore.atlassian.net/browse/MIR-1260).

### CODE REVIEW

**Comment by toKrause** [comment]
Just for comparison:

Inbuild:
image


Generated:
image

File: mir-module/src/main/resources/xslt/generate-svg-by-type.xsl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    <xsl:call-template name="generate-download-svg">
+      <xsl:with-param name="type" select="$value"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <xsl:template name="generate-download-svg">
+    <xsl:param name="type"/>
+
+    <svg id="Ebene_1" data-name="Ebene 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36.88 41">
```
Comment (toKrause) [inline mir-module/src/main/resources/xslt/generate-svg-by-type.xsl:31]: Attributes `id` and `data-name` are unnecessary. Probably just a remnant from whichever editor was used to create the original SVG.

File: mir-module/src/main/resources/xslt/generate-svg-by-type.xsl [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        </style>
+      </defs>
+      <title>
+        <xsl:value-of select="concat('download_', $type)"/>
+      </title>
+
+      <path class="cls-1" d="M10.2,38.43c0,.06,0,0,0,0h0Z"/>
+      <path class="cls-2"
+            d="M29,.5H10.31a2,2,0,0,0-1.37.54,1.79,1.79,0,0,0-.57,1.31V8.31A1,1,0,0,1,9,8.11a1,1,0,0,1,.7.27V2.25a.49.49,0,0,1,.19-.39.61.61,0,0,1,.45-.15L29,1.69l.25,3.69H35c.08,0,.11.4.11.55l0,32.69a.61.61,0,0,1-.19.44.65.65,0,0,1-.45.19h-14a1,1,0,0,1,.2.56,1,1,0,0,1-.29.68H34.45a2,2,0,0,0,1.37-.54,1.8,1.8,0,0,0,.57-1.31L36.36,6A3.21,3.21,0,0,0,35,3.52l-3.31-2.2A5.65,5.65,0,0,0,29,.5Zm1.46,3.88-.15-2.29a2.83,2.83,0,0,1,.67.36l2.83,1.93Z"/>
+      <g id="_18-ecommerce_-_download_digital_item" data-name="18-ecommerce - download digital item">
```
Comment (toKrause) [inline mir-module/src/main/resources/xslt/generate-svg-by-type.xsl:44]: Attributes `id` and `data-name` are unnecessary. Probably just a remnant from whichever editor was used to create the original SVG. The two `g` elements wrapping the single `path` element are also unnecessary.

**Comment by Possommi** [comment]
> Just for comparison:
> 
> Inbuilt: image
> 
> Generated: image

I will try to figure out the original font used and use that one.

File: mir-module/src/main/resources/config/mir/mycore.properties [hunk 3]
```diff
@@ -764,3 +764,7 @@ MIR.CanonicalBaseURL=
 MIR.Sherpa=org.mycore.mir.sherpa.MCRSherpaConfig
 MIR.Sherpa.API.Key=
 MIR.Sherpa.API.URL=https://v2.sherpa.ac.uk/
+
+MCR.ContentTransformer.svg-download.Class      = org.mycore.common.content.transformer.MCRXSLTransformer
+MCR.ContentTransformer.svg-download.Class      = org.mycore.common.content.transformer.MCRXSLTransformer
```
Comment (toKrause) [inline mir-module/src/main/resources/config/mir/mycore.properties:769]: Property `MCR.ContentTransformer.svg-download.Class` appears to be added twice

File: mir-module/src/main/resources/xsl/mir-utils.xsl [hunk 4]
```diff
@@ -101,9 +102,16 @@
                 <xsl:value-of select="concat($baseURL,'images/svg_icons/download_video.svg')"/>
             </xsl:when>
             <xsl:otherwise>
-                <xsl:value-of select="concat($baseURL,'images/svg_icons/download_default.svg')"/>
+                <xsl:choose>
+                    <xsl:when test="string-length($derivateMaindoc) &gt; 0 and string-length($returnId) &gt; 0">
+                        <xsl:variable name="ext" select="substring-after($derivateMaindoc, '.')"/>
```
Comment (toKrause) [inline mir-module/src/main/resources/xsl/mir-utils.xsl:107]: Method `substring-after` is problematic when the filename contains multiple dots, like [`mods2marc-3.7map-base-v.1.0-master.xlsx`](https://www.loc.gov/standards/mods/v3/source/mods2marc-3.7map-base-v.1.0-master.xlsx), which would result in the "extension" `.7map-base-v.1.0-master.xlsx`.

**Suggestion:** Could maybe reuse something existing to obtain the extension. For example, in `mir-video.js.xsl`, `org.apache.commons.io.FilenameUtils#getExtension(String)` is used. Could even be be called without a Xalan-extension:

```
<xsl:variable name="ext" select="document(concat('callJava:org.apache.commons.io.FilenameUtils:getExtension:',$derivateMaindoc))"/>

**Review by toKrause — CHANGES_REQUESTED** [review]

**Review by sebhofmann — APPROVED** [review]

File: mir-module/src/main/resources/xsl/mir-utils.xsl [hunk 5]
```diff
@@ -31,6 +30,8 @@
     <xsl:template name="iconLink">
         <xsl:param name="baseURL"/>
         <xsl:param name="mimeType"/>
+        <xsl:param name="derivateMaindoc"/>
+        <xsl:param name="returnId"/>
```
Comment (toKrause) [inline mir-module/src/main/resources/xsl/mir-utils.xsl:34]: The parameter `returnId` is not used. Possibly a remnant of an earlier version?

**Review by toKrause — CHANGES_REQUESTED** [review]

File: mir-module/src/main/resources/xsl/mir-utils.xsl (hunk 5 above)
Comment (Possommi) [reply mir-module/src/main/resources/xsl/mir-utils.xsl:34]: it is used in line 106

**Review by kkrebs — APPROVED** [review]

File: mir-module/src/main/resources/META-INF/resources/images/svg_icons/download-generic.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
...Asi0QZUP9nFLxRMQPWOMKi6cuNEm0dOr6GYtFgjSN46mNTcqUPAlMQ3p3JW1YBZ50TfM8MrUWM1byTvfLUw1zIpMW+/lgLaURCLTLjJWZJ8K5HyLTPSIztk2jTz+noGwl2Q+LpHdxlhs2E9VeyqXsymowlg2TSTVok2cdFStLNQnoATh67xcX20G8yf801GO5LYCY9P99JvfRzwKYEo4QbQio9Tvnj0g4AAAAAAAAAAAAAAACCT5HYI21adk4dwoQy4EIqbWbXJUwZN4RU7745orykE80TH8C23JxOHo5ljBtCaTPni+VeJE4CZfxJccvAvJ4ETkf/7aYRJpQBN4RU2szqZEJBMG4kAgN4CjEkCk5Jtjs5pHz/BqmQuPDItBxcouxs4YYPc0jHRzoUyphZahJn3/3tA2j4D4oBQlTAKrACDBQIMicAKqCiAr7mBKDqAuciAKQCAASkBNAIAACuCEQJAASFQpR+/m3wFJg5AqREgEqRwAoIBgMBEAGNCIQAApAywZciVAvxLwAAKgBBQEoAjQAA4IpAlAAggEaHNZAdxDChDLgh5lo2z46yGmttq71qnNkoZMJOYbjlZXMrHj18FNYFSP7dz5O4J7kthlr1DmDOfnCSYd4/ZOs3CdP8wJh8cyU2XDwLjzeyYe3/Q01/AG17D017Ym6j6v5DZRMVlRLllWW1T15TUkZxboQpvxjzahhyEvTFqa6oq7BgQ5uVoslUp0pVWs4uLUWajvtEWRiHkcUljY2QRFjiaAKzyFUYKgSB/KAUNwiu3wbbX2yL6clyDUxHhuOG6mBBs6FaUkxkYkoy64JoTGBkBB73MIBYhOnplaefkHsEcw4Q2dorsrRrpnYOzP7uO6RVR6XmEQJffyCGV+V7D5LAdXbAif3e28EFLBkw5X9E4AQodoRrknB/DNJwEP8RIvwDUsq4kzKE7k7BEH/Og1fh/9Y3Ve2Eflo1rp9DDG51lat9HjgM8xseKzHcmGy16dWKZwBTCGlQ4Z8Za4KwGg0mHThKvDqJ1WIChWF6Q2PEOHJp/ONdHMVfBj0/qnWDNSSta+CXgZ39O35SQfHr0K+uJSP+FL7/8QNmN0f4pSu/CjCzmyETVRPJLH4KzU/MEuO9/wSBYAlzmJNKqC2+hQH+Wt6+d1EMfYGFptWXfBam7EO8AgAAAA==) format('woff2');
```
Comment (toKrause) [inline mir-module/src/main/resources/META-INF/resources/images/svg_icons/download-generic.xml:15]: I think it is questionable to include the entire font in each generated SVG image

**Pull request merged**